ipc/cli/src/commands/subnet/join.rs

```diff
@@ -26,18 +27,14 @@ impl CommandLineHandler for JoinSubnet {
             Some(address) => Some(Address::from_str(address)?),
             None => None,
         };
-        let worker_addr = match &arguments.worker_addr {
-            Some(address) => Some(Address::from_str(address)?),
-            None => None,
-        };
-
+        let metadata =
+            base64::engine::general_purpose::STANDARD_NO_PAD.decode(&arguments.metadata)?;
```

Comment:
The only metadata that we currently need is the public key of the validator, right? I think we should have `pub_key` as an argument instead of the whole metadata. It can be really confusing for users otherwise.